build/images/scripts/start_ovs_windows_containerd.ps1

```diff
@@ -0,0 +1,156 @@
+        "$time $Info `n`r" | Tee-Object $InstallLog -Append | Write-Host
+    }
+
+    function CreatePath($Path){
+        if ($(Test-Path $Path)) {
+            mv $Path $($Path + "_bak")
+        }
+        mkdir -p $Path | Out-Null
+    }
+
+    function SetEnvVar($key, $value) {
+        [Environment]::SetEnvironmentVariable($key, $value, [EnvironmentVariableTarget]::Machine)
+    }
+
+    function WaitExpandFiles($Src, $Dest) {
+        Log "Extract $Src to $Dest"
+        Expand-Archive -Path $Src -DestinationPath $Dest | Out-Null
+    }
+
+    function CheckIfOVSInstalled() {
+        if (Test-Path -Path $OVSInstallDir) {
+            Log "$OVSInstallDir already exists, exit OVS installation."
+            exit 1
+        }
+    }
+
+    If (!(Test-Path $OVSDownloadDir)) {
+        mkdir -p $OVSDownloadDir
+    }
+    Log "Downloading OVS package from $OVSDownloadURL to $OVSZip"
+    curl.exe -sLo $OVSZip $OVSDownloadURL
+    If (!$?) {
+        Log "Download OVS failed, URL: $OVSDownloadURL"
+        exit 1
+    }
+
+    if ($CheckFileHash) {
+        $FileHash = Get-FileHash $OVSZip
+        If ($OVSPublishedHash -ne "" -And $FileHash.Hash -ne $OVSPublishedHash) {
+            Log "SHA256 mismatch for OVS download"
+            exit 1
+        }
+    }
+
+    WaitExpandFiles $OVSZip $OVSDownloadDir
+    # Copy OVS package to target dir.
+    Log "Copying OVS package from $OVSDownloadDir\openvswitch to $OVSInstallDir"
+    mv "$OVSDownloadDir\openvswitch" $OVSInstallDir
+    rm $OVSZip
+    # Create log and run dir.
+    $OVS_LOG_PATH = $OVSInstallDir + "\var\log\openvswitch"
+    CreatePath $OVS_LOG_PATH
+    $OVSRunDir = $OVSInstallDir + "\var\run\openvswitch"
+    CreatePath $OVSRunDir
+    $OVSDriverDir = "$OVSInstallDir\driver"
+
+    # Install OVS driver certificate.
+    $DriverFile="$OVSDriverDir\OVSExt.sys"
+    if ($ImportCertificate) {
+        $CertificateFile = "$OVSDriverDir\package.cer"
+        if (!(Test-Path $CertificateFile)) {
+            $ExportType = [System.Security.Cryptography.X509Certificates.X509ContentType]::Cert;
+            $Cert = (Get-AuthenticodeSignature $DriverFile).SignerCertificate;
+            [System.IO.File]::WriteAllBytes($CertificateFile, $Cert.Export($ExportType));
+        }
+        Log "Installing OVS driver certificate."
+        Import-Certificate -FilePath "$CertificateFile" -CertStoreLocation cert:\LocalMachine\TrustedPublisher
+        Import-Certificate -FilePath "$CertificateFile" -CertStoreLocation cert:\LocalMachine\Root
+    }
+
+    # Install Microsoft Visual C++ Redistributable Package.
+    if (Test-Path $OVSInstallDir\redist) {
+        Log "Installing Microsoft Visual C++ Redistributable Package."
+        $RedistFiles = Get-ChildItem "$OVSInstallDir\redist" -Filter *.exe
+        $RedistFiles | ForEach-Object {
+            Log "Installing $_"
+            Start-Process -FilePath $_.FullName -Args '/install /passive /norestart' -Verb RunAs -Wait
+        }
+    }
```

Comment:
V c++ installation might not be needed as vcruntime140.dll is already copied in windows image
can remove these lines !